orchestra/static/orchestra/timing/timer/directives.js

```diff
@@ -77,21 +91,35 @@
             .then(function(response) {
               timeEntries.addEntry(response.data);
               resetTimer();
+              if (!response.data.description) {
+                if (confirm("You didn't enter a description for this time entry. Would you like to view your timecard and edit it there?")) {
+                  workTimer.viewTimecard();
+                }
+              }
             }, function() {
                 alert('Could not stop timer');
             });
           };
 
-          workTimer.updateDescription = function() {
-            var url = '/orchestra/api/interface/timer/description/';
-            $http.post(url, {
-              description: workTimer.description,
-            })
+          workTimer.updateTimer = function() {
+            var url = '/orchestra/api/interface/timer/update/';
+            var assignment_id = workTimer.task ? workTimer.task.assignment_id : null;
+            var data = {
+              assignment: assignment_id,
+              description: workTimer.description
+            };
+            $http.post(url, data)
             .catch(function() {
                 alert('Could not update timer description');
             });
           };
 
+          $scope.$watch('workTimer.task', function(newVal, oldVal) {
```

Comment:
Yes—this is what updates the task server side when a new task is selected in the interface